share/shwap/p2p/bitswap/bitswap.go

```diff
@@ -0,0 +1,136 @@
+package bitswap
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ipfs/boxo/bitswap/client"
+	"github.com/ipfs/boxo/bitswap/network"
+	"github.com/ipfs/boxo/bitswap/server"
+	"github.com/ipfs/boxo/blockstore"
+	delay "github.com/ipfs/go-ipfs-delay"
+	routinghelpers "github.com/libp2p/go-libp2p-routing-helpers"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/protocol"
+)
+
+// Client constants
+const (
+	// simulateDontHaves emulates DONT_HAVE message from a peer after 5 second timeout.
+	// This protects us from unresponsive/slow peers.
+	// TODO(@Wondertan): PR to bitswap to make this timeout configurable
+	//  Higher timeout increases the probability of successful reconstruction
+	simulateDontHaves = true
+	// providerSearchDelay defines the initial delay before Bitswap client starts aggressive
+	// broadcasting of WANTs to all the peers. We offset this for longer than the default to minimize
+	// unnecessary broadcasting as in most cases we already have peers connected with needed data on
+	// a new request.
+	providerSearchDelay = time.Second * 10
+	// rebroadcastDelay is similar to the providerSearchDelay, but it targets DHT/ContentRouting
+	// peer discovery and a gentle broadcast of a single random live WANT to all connected peers.
+	// Considering no DHT usage and broadcasting configured by providerSearchDelay, we set
+	// rebroadcastDelay to max value, effectively disabling it
+	rebroadcastDelay = 1<<63 - 1
+)
+
+// Server constants
+const (
+	// providesEnabled dictates Bitswap Server not to provide content to DHT/ContentRouting as we don't use it
+	providesEnabled = false
+	// sendDontHaves prevents Bitswap Server from sending DONT_HAVEs while keeping peers on hold instead:
+	//  * Clients simulate DONT_HAVEs after timeout anyway
+	//  * Servers may not have data immediately and this gives an opportunity to subscribe
+	//  * This is necessary for reconstruction. See https://github.com/celestiaorg/celestia-node/issues/732
+	sendDontHaves = false
```

Comment:
In other words, turning off don't haves turns bitswap into async/subscribe mod. It is important for cases when we fallback to reconstruction to allow FN to act as samples relay